app/src/main/java/org/kiwix/kiwixmobile/nav/destination/library/OnlineLibraryFragment.kt

```diff
@@ -398,6 +400,31 @@ class OnlineLibraryFragment : BaseFragment(), FragmentActivityExtensions {
     }
   }
 
+  private fun checkNotificationPermission(): Boolean {
+    return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+      hasPermission(POST_NOTIFICATIONS).also { permissionGranted ->
+        if (!permissionGranted) {
+          if (shouldShowRationale(POST_NOTIFICATIONS)) {
+            alertDialogShower.show(
+              KiwixDialog.NotificationPermissionRationale,
+              {
+                requestPermission(
+                  Manifest.permission.POST_NOTIFICATIONS,
+                  REQUEST_POST_NOTIFICATION_PERMISSION
+                )
+              }
+            )
+          } else {
+            alertDialogShower.show(
+              KiwixDialog.NotificationPermissionRationale,
+              requireActivity()::navigateToAppSettings
+            )
```

Comment:
Perhaps a better condition will be if the device is greater than 13 and the user hasn't given permission yet. Show the permission again, if the device is lower or equal to 12 then navigate to the settings because it'll show once in the lower than 12 devices. What do you think?